acj/api/login.py

```diff
@@ -25,9 +25,21 @@ def login():
     else:
         permissions = authenticate(user)
 
-        if sess.get('LTI') and sess.get('oauth_create_user_link'):
+        if sess.get('LTI') and sess.get('oauth_create_user_link') and sess.get('lti_user') and sess.get('lti_context'):
             lti_user = LTIUser.query.get_or_404(sess['lti_user'])
             lti_user.acj_user_id = user.id
+            lti_context = LTIContext.query.get_or_404(sess['lti_context'])
+            if lti_context.is_linked_to_course():
+                user_course = UserCourse.query \
+                .filter_by(
+                    user_id=user.id,
+                    course_id=lti_context.acj_course_id
+                ) \
+                .one_or_none()
+                if user_course is None:
+                    # create new enrollment
+                    new_user_course = UserCourse(user_id=user.id, course_id=lti_context.acj_course_id, course_role=lti_user.system_role)
```

Comment:
you need to load lti_user_resource_link for course role since system_role != course_role
`course_role=lti_user_resource_link.course_role`